QuantLib/ql/termstructures/yield/nonlinearfittingmethods.cpp

```diff
@@ -213,6 +222,35 @@ namespace QuantLib {
         }
         return d;
     }
+	
+	SpreadFittingMethod::SpreadFittingMethod(shared_ptr<FittingMethod> method,
+                        Handle<YieldTermStructure> discountCurve)
+    : FittedBondDiscountCurve::FittingMethod(method ? method->constrainAtZero() : true, method ? method->weights() : Array(), 
+											 method ? method->optimizationMethod() : boost::shared_ptr<OptimizationMethod>()),
+      method_(method), discountingCurve_(discountCurve) {
+		QL_REQUIRE(!method, "Fitting method is empty");
```

Comment:
You're negating `method`, so this actually requires that the method is null. It should be `QL_REQUIRE(method, ...)` instead.